spring/src/main/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerFactoryProcessor.java

```diff
@@ -121,6 +61,75 @@ public int getOrder() {
   @Override
   public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry)
       throws BeansException {
-    this.beanDefinitionRegistry = registry;
+    if (registry instanceof BeanFactory) {
+      for (String name : registry.getBeanDefinitionNames()) {
+        Class<?> cls = ((BeanFactory) registry).getType(name);
+        if (cls != null) {
+          KubernetesInformers kubernetesInformers =
+              AnnotatedElementUtils.getMergedAnnotation(cls, KubernetesInformers.class);
+          if (kubernetesInformers != null && kubernetesInformers.value().length > 0) {
+            for (KubernetesInformer kubernetesInformer : kubernetesInformers.value()) {
+              registerInformer(registry, kubernetesInformer);
+              registerLister(registry, kubernetesInformer);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void registerInformer(
+      BeanDefinitionRegistry registry, KubernetesInformer kubernetesInformer) {
+    RootBeanDefinition informerBean =
+        (RootBeanDefinition)
+            BeanDefinitionBuilder.rootBeanDefinition(SharedInformer.class).getBeanDefinition();
+    informerBean.setInstanceSupplier(
+        () -> informer(kubernetesInformer.apiTypeClass(), kubernetesInformer));
+    ResolvableType informerType =
+        ResolvableType.forClassWithGenerics(
+            SharedInformer.class, kubernetesInformer.apiTypeClass());
+    informerBean.setTargetType(informerType);
+    registry.registerBeanDefinition(kubernetesInformer.apiTypeClass().getName(), informerBean);
+  }
+
+  private void registerLister(
+      BeanDefinitionRegistry registry, KubernetesInformer kubernetesInformer) {
+    RootBeanDefinition listerBean =
+        (RootBeanDefinition)
+            BeanDefinitionBuilder.rootBeanDefinition(Lister.class).getBeanDefinition();
+    listerBean.setInstanceSupplier(
+        () -> lister(kubernetesInformer.apiTypeClass(), kubernetesInformer));
+    ResolvableType listerType =
+        ResolvableType.forClassWithGenerics(Lister.class, kubernetesInformer.apiTypeClass());
+    listerBean.setTargetType(listerType);
+    registry.registerBeanDefinition(listerType.toString(), listerBean);
+  }
+
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  private <T extends KubernetesObject> Lister<T> lister(
+      Class<T> type, KubernetesInformer kubernetesInformer) {
```

Comment:
neither args are used. what's the intention?

Reply:
See below.